app/bundles/ProjectBundle/Tests/Functional/Controller/ProjectEntityLookupLimitTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\ProjectBundle\Tests\Functional\Controller;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\EmailBundle\Model\EmailModel;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\HttpFoundation\Response;
+
+final class ProjectEntityLookupLimitTest extends MauticMysqlTestCase
+{
+    private const LOOKUP_CHOICE_LIST_URL = '/s/ajax?action=project:getLookupChoiceList';
+
+    /**
+     * Create 2000 test emails.
+     */
+    private function createTestEmails(): void
+    {
+        /** @var EmailModel $emailModel */
+        $emailModel = self::getContainer()->get(EmailModel::class);
+
+        for ($i = 1; $i <= 2000; ++$i) {
+            $email = new Email();
+            $email->setName('Common Autocomplete Email '.$i);
+            $email->setSubject('Subject '.$i);
+            $email->setEmailType('template');
+            $email->setTemplate('blank');
+            $emailModel->saveEntity($email);
+        }
+    }
+
+    /**
+     * Test: AJAX search with a common keyword
+     * Must return exactly 1000 results.
+     */
+    public function testAjaxSearchReturnsExactly1000Results(): void
+    {
+        // Arrange: seed 2000 emails
+        $this->createTestEmails();
+
+        // Act: trigger AJAX lookup endpoint
+        $this->client->request('GET', self::LOOKUP_CHOICE_LIST_URL, [
+            'entityType' => 'email',
+            'filter'     => 'Common',
+        ]);
+
+        // Assert: response is valid
+        $this->assertResponseIsSuccessful();
```

Comment:
```suggestion
        $this->assertResponseIsSuccessful();
```
I think the comment isn't necessary. The method name says it all.

Reply:
This is fixed.